interactors/wallet.go

```diff
@@ -226,20 +228,25 @@ func (w *wallet) LoadPrivateKeyFromJsonFile(filename string, password string) ([
 	privateKey := make([]byte, len(cipherText))
 	stream.XORKeyStream(privateKey, cipherText)
 
-	if len(privateKey) > 32 {
-		privateKey = privateKey[:32]
+	if key.Kind != mnemonicKind { // wallets with the old JSON format
+		if len(privateKey) > 32 {
+			privateKey = privateKey[:32]
+		}
+		address, errGetAddr := w.GetAddressFromPrivateKey(privateKey)
+		if errGetAddr != nil {
+			return nil, errGetAddr
+		}
+
+		isSameAccount := hex.EncodeToString(address.AddressBytes()) == key.Address &&
+			address.AddressAsBech32String() == key.Bech32
+		if !isSameAccount {
+			return nil, ErrDifferentAccountRecovered
+		}
+	} else {
+		// on the new JSON versions, the mnemonic is encrypted, instead of the private key. Therefore, get the private key from mnemonic
+		privateKey = w.GetPrivateKeyFromMnemonic(data.Mnemonic(privateKey), 0, 0)
```

Comment:
Perhaps we can rename the variable `privateKey` defined on line 228 to `decryptedData`?

While here, indeed, the result of `GetPrivateKeyFromMnemonic()` would be kept in a variable `privateKey`, but a different one.

Oh, also - `secretKey`, instead of `privateKey`.